Sources/Fluent/Database/Database.swift

```diff
@@ -18,15 +18,46 @@ public final class Database: Executor {
     /// one for each thread
     public let threadConnectionPool: ThreadConnectionPool
 
+    /// The string value for the
+    /// default identifier key.
+    ///
+    /// The `idKey` will be used when
+    /// `Model.find(_:)` or other find
+    /// by identifier methods are used.
+    ///
+    /// This value is overriden by
+    /// entities that implement the
+    /// `Entity.idKey` static property.
+    public var idKey: String
+
+    /// The default type for values stored against the identifier key.
+    ///
+    /// The `idType` will be accessed by those Entity implementations
+    /// which do not themselves implement `Entity.idType`.
+    public var idType: IdentifierType
+
     /// Creates a `Database` with the supplied
     /// `Driver`. This cannot be changed later.
     public init(_ driver: Driver, maxConnections: Int = 128) {
+        idKey = driver.idKey
+        idType = driver.idType
+        shouldLog = false
+        logs = []
+
         threadConnectionPool = ThreadConnectionPool(
             makeConnection: driver.makeConnection,
             maxConnections: maxConnections // some number larger than the max threads
         )
         self.driver = driver
     }
+
+    /// If set to true, the database will store
+    /// all queries ran in the `logs` property.
+    public var shouldLog: Bool
+
+    /// If `shouldLog` is set to true, all queries
+    /// performed by this database will be stored here.
+    public var logs: [Log]
```

Comment:
Yeah I was thinking about that. Ideally `shouldLog` would only be turned on in testing environments but I could see people forgetting about it.

Maybe something like `var didLog: (Log) -> ()`. Every time the database does a query it passes the log to that method. The end user could just print the log if that's what they are looking for, or they could store it to their own array if that's what they are looking for.